resources/jscomposition/cases/casesDetail/edit.js

```diff
@@ -1,8 +1,366 @@
+      const status = {
+        ACTIVE: this.request.created_at,
+        COMPLETED: this.request.completed_at,
+        CANCELED: this.request.updated_at,
+        ERROR: this.request.updated_at,
+      };
+
+      return status[this.request.status.toUpperCase()];
+    },
+    statusLabel() {
+      const status = {
+        ACTIVE: this.$t("In Progress"),
+        COMPLETED: this.$t("Completed"),
+        CANCELED: this.$t("Canceled"),
+        ERROR: this.$t("Error"),
+      };
+
+      return status[this.request.status.toUpperCase()];
+    },
+    requestBy() {
+      return [this.request.user];
+    },
+    panCommentInVueOptionsComponents() {
+      return "pan-comment" in Vue.options.components;
+    },
+  },
+  mounted() {
+    this.packages = window.ProcessMaker.requestShowPackages;
+    this.listenRequestUpdates();
+    this.cleanScreenButtons();
+    this.editJsonData();
+  },
+  methods: {
+    switchTab(tab) {
+      this.activeTab = tab;
+      if (tab === "overview") {
+        this.isObjectLoading = true;
+      }
+      ProcessMaker.EventBus.$emit("tab-switched", tab);
+    },
+    switchTabInfo(tab) {
+      this.showInfo = !this.showInfo;
+      if (window.Intercom) {
+        window.Intercom("update", { hide_default_launcher: tab === "comments" });
+      }
+    },
+    onLoadedObject() {
+      this.isObjectLoading = false;
+    },
+    requestStatusClass(status) {
+      const bubbleColor = {
+        active: "text-success",
+        inactive: "text-danger",
+        error: "text-danger",
+        draft: "text-warning",
+        archived: "text-info",
+        completed: "text-primary",
+      };
+      return `fas fa-circle ${bubbleColor[status.toLowerCase()]} small`;
+    },
+    // Data editor
+    updateRequestData() {
+      const data = JSON.parse(this.jsonData);
+      ProcessMaker.apiClient.put(`requests/${this.requestId}`, {
+        data: data,
+      }).then(() => {
+        this.fieldsToUpdate.splice(0);
+        ProcessMaker.alert(this.$t("The request data was saved."), "success");
+      });
+    },
+    saveJsonData() {
+      try {
+        const value = JSON.parse(this.jsonData);
+        this.updateRequestData();
+      } catch (e) {
+        // Invalid data
+      }
+    },
+    editJsonData() {
+      this.jsonData = JSON.stringify(this.data, null, 4);
+    },
+    /**
+     * Refresh the Request details.
+     *
+     */
+    refreshRequest() {
+      this.$refs.pending.fetch();
+      this.$refs.completed.fetch();
+      ProcessMaker.apiClient.get(`requests/${this.requestId}`, {
+        params: {
+          include: "participants,user,summary,summaryScreen",
+        },
+      }).then((response) => {
+        for (const attribute in response.data) {
+          this.updateModel(this.request, attribute, response.data[attribute]);
+        }
+        this.refreshTasks++;
+      });
+    },
+    /**
+     * Update a model property.
+     *
+     */
+    updateModel(obj, prop, value, defaultValue) {
```

Comment:
Where do we use this method?
 Is necesary this method in the new implementation?